src/Checkers/LicenseChecker.js

```diff
@@ -0,0 +1,281 @@
+
+    if (output) {
+      return new LicenseCheckerWarning(output.message, offset + output.line, filepath, output.symbol);
+    }
+
+    return false;
+  }
+
+  _compareWithLicense(checkedLicense) {
+    const originalLicense = fs.readFileSync(LICENSE_FILE_PATH, 'utf-8');
+    checkedLicense = checkedLicense.replace(/\d\d\d\d(\-\d\d\d\d)?/, '');
+    return this._compareLicenses(checkedLicense, originalLicense);
+  }
+
+  _compareLicenses(checkedLicense, originalLicense) {
+    let checkedLicIndex = 0;
+    let originalLicIndex = 0;
+    let lineNum = 1;
+    let lastLB = 0;
+
+    for (; (checkedLicIndex < checkedLicense.length) && (originalLicIndex < originalLicense.length); checkedLicIndex++, originalLicIndex++) {
+      let checkedSpace = false;
+      let originalSpace = false;
+      while ( (checkedLicIndex < checkedLicense.length) && (/\s/.test(checkedLicense[checkedLicIndex]))) {
+        if (checkedLicense[checkedLicIndex] == '\n') {
+          lineNum++;
+          lastLB = checkedLicIndex;
+        }
+        checkedSpace = true;
+        checkedLicIndex++;
+      }
+
+      while ( (originalLicIndex < originalLicense.length) && (/\s/.test(originalLicense[originalLicIndex]))) {
+        originalSpace = true;
+        originalLicIndex++;
+      }
+
+      if (originalSpace != checkedSpace) {
+        let message = '';
+        if (originalSpace) {
+          message = 'missing space in license';
+        } else {
+          message = 'extra space in license';
+        }
+       return {
+          message : message,
+          line : lineNum,
+          symbol :checkedLicIndex - lastLB - 1
+        };
+      }
+
+      // check if we doesn't reach end of licenses
+      if ( !((checkedLicIndex < checkedLicense.length) && (originalLicIndex < originalLicense.length))) break;
+
+      if (checkedLicense[checkedLicIndex] != originalLicense[originalLicIndex]) {
+        const firstWord = this._findNearestWord(checkedLicense, checkedLicIndex);
+        const secondWord = this._findNearestWord(originalLicense, originalLicIndex);
+        return {
+          message : `expected "${secondWord}", but find "${firstWord}" in license header`,
+          line : lineNum,
+          symbol :checkedLicIndex - lastLB
+        };
+      }
+
+    }
+    let tail;
+    // it is ok if we have another comments on the next line
+    // we decrease checkedLicIndex because we skipped \n
+    if (tail = this._isStringHasEnd(checkedLicense, checkedLicIndex - 1, /^(\s*\n[\s|\S]*)?$/)) {
+      return {
+        message : `unexpected end of license "${tail}"`,
+        line : checkedLicIndex - lastLB
+      };
+    }
+
+    if (tail = this._isStringHasEnd(originalLicense, originalLicIndex, /^(\s+)?$/)) {
+      return {
+        message : `missing end of license "${tail}"`,
+        line : checkedLicIndex - lastLB
+      };
+    }
+    return false;
+  }
+
+  _isStringHasEnd(str, index, acceptableEnd) {
```

Comment:
mm, `endsWith`? This is very odd method! I would just get rid of it.

it's a single line: `suffix.test(str.substr(index))`